internal/workflow_testsuite.go

```diff
@@ -543,6 +544,143 @@ func (e *TestWorkflowEnvironment) OnUpsertMemo(attributes interface{}) *MockCall
 	return e.wrapWorkflowCall(call)
 }
 
+// OnNexusOperation setup a mock call for Nexus operation.
+// Parameter service must be Nexus service (*nexus.Service) or service name (string).
+// Parameter operation must be Nexus operation (nexus.RegisterableOperation) or operation name (string).
+// You must call Return() with appropriate parameters on the returned *MockCallWrapper instance. The
+// supplied parameters to Return() call must be an instance of nexus.HandlerStartOperationResult[T]
+// and an error. If your mock returns nexus.HandlerStartOperationResultAsync, you need to register the
+// completion of the async operation by calling RegisterNexusAsyncOperationCompletion.
+// Example: assume the Nexus operation you want to mock is as follows:
+//
+//	type (
+//		HelloInput struct {
+//			Message string
+//		}
+//		HelloOutput struct {
+//			Message string
+//		}
+//	)
+//
+//	var HelloOperation = temporalnexus.NewWorkflowRunOperation(
+//		"hello-operation",
+//		HelloHandlerWorkflow,
+//		func(ctx context.Context, input HelloInput, options nexus.StartOperationOptions) (client.StartWorkflowOptions, error) {
+//			return client.StartWorkflowOptions{}, nil
+//		},
+//	)
+//
+//	func HelloHandlerWorkflow(_ workflow.Context, input HelloInput) (HelloOutput, error) {
+//		return HelloOutput{Message: "Hello " + input.Message}, nil
+//	}
+//
+// Then, you can mock workflow.NexusClient.ExecuteOperation as follows:
+//
+//	t.OnNexusOperation(
+//		service,
+//		HelloOperation,
+//		HelloInput{Message: "Temporal"},
+//		mock.Anything,
+//	).Return(
+//		&nexus.HandlerStartOperationResultAsync{
+//			OperationID: "hello-operation-id",
+//		},
+//		nil,
+//	)
+//
+//	t.RegisterNexusAsyncOperationCompletion(
+//		service,
+//		HelloOperation.Name(),
+//		"hello-operation-id",
+//		HelloOutput{Message: "Hello Temporal"},
+//		nil,
+//		1*time.Second,
+//	)
```

Comment:
In addition to this, maybe we could have helper functions to mock sync and async operations. Example: for async operation, we could have something like this:
```
func (e *TestWorkflowEnvironment) MockNexusAsyncOperation(
	service any,
	operation any,
	input any,
	options any,
	result any,
	err error,
	delay time.Duration,
) *MockCallWrapper {
	operationID := uuid.New() // maybe also input?
	m := e.OnNexusOperation(service, operation, input, options).Return(
		&nexus.HandlerStartOperationResultAsync{
			OperationID: operationID,
		},
		nil,
	)
	e.RegisterNexusAsyncOperationCompletion(service, operation, operationID, result, err, delay)
	return m
}
```

And we could also have something similar for the sync which basically abstract the returned value type `nexus.HandlerStartOperationResultSync`.

Reply:
Alternatively, if a user sets `After` on the returned `MockCallWrapper`, it sort of makes the operation async already (with the exception of generating a separate started event).
Can you config `After` worker with the `MockCallWrapper` returned here?

Reply:
Yes, you can, but as you said, it would not represent the actual behavior of async operation.